dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -317,6 +315,56 @@ def _CheckSkippedFlows(self) -> None:
               client_id, flow_id, self.reason
         ))
 
+  def _DownloadArchive(
+    self, grr_flow: Client.Flow, flow_output_dir: str
+  ) -> None:
+    output_file_path = os.path.join(self.output_path, f"{grr_flow.flow_id}.zip")
+    file_archive = grr_flow.GetFilesArchive()
+    file_archive.WriteToFile(output_file_path)
+    with zipfile.ZipFile(output_file_path) as archive:
+      archive.extractall(path=flow_output_dir)
+    os.remove(output_file_path)
+
+  def _DownloadBlobs(
+    self, client: Client, pathspecs: Client.Flow, flow_output_dir: str
+  ) -> None:
+    for pathspec in pathspecs:
+      if pathspec.nested_path.pathtype == pathspec.nested_path.NTFS:
+        vfspath = f"fs/ntfs{pathspec.path}{pathspec.nested_path.path}"
+      else:
+        vfspath = re.sub("^([a-zA-Z]:)?/(.*)$", "fs/os/\\1/\\2", pathspec.path)
+      filename = os.path.basename(vfspath)
+      base_dir = os.path.join(flow_output_dir, os.path.dirname(vfspath))
+      os.makedirs(base_dir, exist_ok=True)
+
+      f = client.File(vfspath)
+      self.logger.debug(f"Downloading blob {filename} from {vfspath}")
+      try:
+        with open(os.path.join(base_dir, filename), "wb") as out:
+          f.GetBlobWithOffset(0).WriteToStream(out)
+      except grr_errors.ResourceNotFoundError as e:
+        self.logger.warning(
+          f"Failed to download blob {filename} from {vfspath}: {e}"
+        )
+
+  def _DownloadTimeline(
+    self,
+    client: Client,
+    grr_flow: Client.Flow,
+    flow_output_dir: str,
+  ) -> str:
+    final_bodyfile_path = os.path.join(
+      flow_output_dir, f"{grr_flow.flow_id}_timeline.body"
+    )
+    ntfs_inodes = client.data.os_info.system.lower() == "windows"
+    timeline = grr_flow.GetCollectedTimelineBody(
+      timestamp_subsecond_precision=True,
+      inode_ntfs_file_reference_format=ntfs_inodes,
+      backslash_escape=True,
+    )
+    timeline.WriteToFile(final_bodyfile_path)
+    return final_bodyfile_path
+
   # TODO: change object to more specific GRR type information.
```

Comment:
This looks like it no longer applies, as the type hint here has a more up-to-date definition - while you're here, please update the docstring and remove the todo